src/R/Components/Impl/Controller/IMenuCommand.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+
+namespace Microsoft.R.Components.Controller {
+    /// <summary>
+    /// An object that implements a single command. 
+    /// </summary>
+    public interface IMenuCommand {
```

Comment:
The nice thing about having a new interface that doesn't take IDs at all is that it is then not tied to the OLE/VS command system in any way - and so reusable in VSCode etc in theory (obviously there are dependencies on VS editor components etc that make it infeasible for many commands - but those that only use `IRSession` and/or `IRInteractiveWorkflow` should be readily portable).